### PR TITLE
feature: generate proof in update

### PR DESCRIPTION
**Pull request by dingo-emperor**

### CODE REVIEW

File: ord/stateless/queue.go [hunk 1]
```diff
@@ -224,13 +236,142 @@ func NewQueues(getter getter.OrdGetter, header *Header, queryHash bool, startHei
...
+		ExtStatus:  es,
+		PoaStems:   poas,
+		Keys:       keys,
+		PreValues:  pe.Vals,
+		PostValues: postvals,
+	}
+	return proof, nil
+}
+
+func (queue *Queue) VerifyProof() bool {
```
Comment (archidoge0) [inline ord/stateless/queue.go:327]: Please move the `VerifyProof` function to the unit tests `Test_NewProof`.
It doesn't perform the actual `VerifyProof` semantic.

File: ord/stateless/queue.go [hunk 2]
```diff
@@ -224,13 +236,142 @@ func NewQueues(getter getter.OrdGetter, header *Header, queryHash bool, startHei
...
+	}
+	vProofBytes, err := vProof.MarshalJSON()
+	if err != nil {
+		return false
+	}
+	finalproof := base64.StdEncoding.EncodeToString(vProofBytes[:])
+	log.Println("VerifyProof finalproof:", finalproof)
+	return finalproof == queue.RollingbackProof()
+}
+func (queue *Queue) RollingbackProof() string {
```
Comment (archidoge0) [inline ord/stateless/queue.go:345]: No longer need this function beyond the `Test_NewProof`?

File: ord/stateless/types.go [hunk 3]
```diff
@@ -66,8 +66,9 @@ type LightHeader struct {
 }
 
 type Queue struct {
-	Header  *Header
-	History [ord.BitcoinConfirmations]DiffState
+	Header         *Header
+	History        [ord.BitcoinConfirmations]DiffState
+	LastStateProof *verkle.Proof
```
Comment (archidoge0) [inline ord/stateless/types.go:71]: You may apply the `LastStateProof` to the `apis.go`.

**Review by archidoge0 — CHANGES_REQUESTED** [review]

File: ord/stateless/queue.go (hunk 2 above)
Comment (dingo-emperor) [reply ord/stateless/queue.go:345]: Yes, but seems I cannot define a new method of a type outside of its local file.

Comment (dingo-emperor) [reply ord/stateless/queue.go:345]: Of course I can also write it in another way other than defining a new method for queue

**Review by archidoge0 — APPROVED** [review]
LGTM

**Pull request merged**